nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -732,8 +733,73 @@ impl StringPrototype {
         Ok(String::from_str(agent, substring).into_value())
     }
 
-    fn split(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    fn split(agent: &mut Agent, this_value: Value, args: ArgumentsList) -> JsResult<Value> {
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        let o = require_object_coercible(agent, this_value)?;
+        // 2. Let S be ? ToString(O).
+        let s = to_string(agent, o)?;
+
+        // 3. Let isRegExp be ? IsRegExp(searchString).
+        // 4. If isRegExp is true, throw a TypeError exception.
+        // TODO
+
+        // 5. Let separatorStr be ? ToString(separatorString).
+        let separator_value = args.get(0);
+
+        // 6. If separator is undefined, return an array with this.
```

Comment:
**issue (nonblocking):** This should be spec step 7, and this is also not the spec text.

I won't block this PR on this, but in the future please leave the spec text in the code mostly as-is. Commenting on the spec text with eg. "Note: ..." is fine. Moving spec text and steps around is also fine as long as it doesn't change any observable features of the function. Oftentimes leaving a note about these sort of reorderings is a good idea.

But rewriting the spec text in your own words, and then in the process having it turn into just your custom algorithm with steps outlined is not a good thing. If someone else comes to look at this function and tries to figure out how this corresponds to the spec algorithm, they'll have very little idea what is going on. Maybe this is written based on some old spec text that is no longer available online? And in that case, what has changed? Presumably the algorithm has been changed in meaningful ways and then this old code written to the old spec is no longer valid; but how?

It's better to abide by the spec text as much as possible, and only make additions (fast paths) and minor reorderings where it makes sense. That way a future reader (which may be you yourself) has an easier time understanding what is being done and why, and if the spec changes they'll also know what they need to change to fix the code.

Reply:
Sorry, I didn't realize these comments were spec comments. I though they were simple comments to follow easier what the code does.
I'll take a look to that.